sdk/js/src/token_bridge/__tests__/sui-integration.ts

```diff
@@ -69,38 +67,75 @@ afterAll(async () => {
-      owner: suiAddress,
-      options: { showContent: true, showType: true },
-    });
-    console.log(JSON.stringify(res.data, null, 2));
-    const coins = res.data.filter((o) =>
-      (o.data?.type ?? "").includes("COIN_8")
+    // // Get COIN_8 coin type
+    // const res = await suiProvider.getOwnedObjects({
+    //   owner: suiAddress,
+    //   options: { showContent: true, showType: true },
+    // });
+    // console.log(JSON.stringify(res.data, null, 2));
+    // const coins = res.data.filter((o) =>
+    //   (o.data?.type ?? "").includes("COIN_8")
+    // );
+    // expect(coins.length).toBe(1);
+    // const coin8 = coins[0];
+    // const coin8Type = getInnerType(getMoveObjectType(coin8) ?? "");
+    // const coin8TreasuryCapObjectId = coin8.data?.objectId;
+    // assertIsNotNullOrUndefined(coin8Type);
+    // assertIsNotNullOrUndefined(coin8TreasuryCapObjectId);
+
+    // Attest on Sui
+    // const suiAttestTxPayload = await attestFromSui(
+    //   suiProvider,
+    //   SUI_TOKEN_BRIDGE_ADDRESS,
+    //   SUI_CORE_STATE_OBJECT_ID,
+    //   SUI_TOKEN_BRIDGE_STATE_OBJECT_ID,
+    //   coin8Type,
+    //   0
+    // );
+    // const suiAttestTxRes = await executeTransactionBlock(
+    //   suiSigner,
+    //   suiAttestTxPayload
+    // );
+    // expect(suiAttestTxRes.effects?.status.status).toBe("success");
+
+    // Start create wrapped on Sui
+    const MOCK_VAA =
+      "0100000000010080366065746148420220f25a6275097370e8db40984529a6676b7a5fc9feb11755ec49ca626b858ddfde88d15601f85ab7683c5f161413b0412143241c700aff010000000100000001000200000000000000000000000000000000000000000000000000000000deadbeef000000000150eb23000200000000000000000000000000000000000000000000000000000000beefface00020c424545460000000000000000000000000000000000000000000000000000000042656566206661636520546f6b656e0000000000000000000000000000000000";
+    const suiPrepareRegistrationTxPayload = await createWrappedOnSuiPrepare(
+      "DEVNET",
+      SUI_CORE_BRIDGE_ADDRESS,
+      SUI_TOKEN_BRIDGE_ADDRESS,
+      Buffer.from(MOCK_VAA, "hex"),
+      suiAddress
     );
-    expect(coins.length).toBe(1);
+    const suiPrepareRegistrationTxRes = await executeTransactionBlock(
+      suiSigner,
+      suiPrepareRegistrationTxPayload
+    );
+    expect(suiPrepareRegistrationTxRes.effects?.status.status).toBe("success");
 
-    const coin8 = coins[0];
-    const coin8Type = getInnerType(getMoveObjectType(coin8) ?? "");
-    const coin8TreasuryCapObjectId = coin8.data?.objectId;
-    assertIsNotNullOrUndefined(coin8Type);
-    assertIsNotNullOrUndefined(coin8TreasuryCapObjectId);
+    // Complete create wrapped on Sui
+    const publishEvents = getPublishedObjectChanges(
+      suiPrepareRegistrationTxRes
+    );
+    expect(publishEvents.length).toBe(1);
 
-    const suiAttestTxPayload = await attestFromSui(
+    const coinPackageId = publishEvents[0].packageId;
+    const coinType = `${coinPackageId}::coin::COIN`;
```

Comment:
Should there be a utility method to create a wrapped coin type? For example, someone calls `getWrappedCoinType(coinPackageId)` so an integrator doesn't have to think about this too much.

If we feel that generating this wrapped coin type is only relevant internally for the SDK, maybe we just pass the `coinPackageId` to `createWrappedOnSui` instead of the coin type.

We may want other utility methods to generate types, too. Not sure where these would go.

Reply:
good thinking, I think the documentation for such a method would be especially helpful for integrators